inginious/frontend/pages/course_admin/utils.py

```diff
@@ -51,26 +51,106 @@ def get_course_and_check_rights(self, courseid, taskid=None, allow_all_staff=Tru
             raise web.notfound()
 
 
-class INGIniousSubmissionAdminPage(INGIniousAdminPage):
+class INGIniousSubmissionsAdminPage(INGIniousAdminPage):
     """
-    An INGIniousAdminPage containing some common methods between download/replay pages
+    An INGIniousAdminPage containing some common methods for querying submissions
     """
 
+    def get_course_params(self, course, params):
+        users = self.get_users(course)
+        audiences = self.user_manager.get_course_audiences(course)
+        tasks = course.get_tasks()
+
+        tutored_audiences = [str(audience["_id"]) for audience in audiences if
+                             self.user_manager.session_username() in audience["tutors"]]
+        tutored_users = []
+        for audience in audiences:
+            if self.user_manager.session_username() in audience["tutors"]:
+                tutored_users += audience["students"]
+
+        limit = params.get("limit", 50) if params.get("limit", 50) > 0 else 50
+
+        return users, tutored_users, audiences, tutored_audiences, tasks, limit
+
+    def get_users(self, course):
+        user_ids = self.user_manager.get_course_registered_users(course)
+        users = {user: self.user_manager.get_user_realname(user) for user in user_ids}
```

Comment:
We may still have some users with no name. It'd be worth replacing this by ``self.user_manager.get_user_realname(user) or ''`` here.

Reply:
yes thanks I will do it